ccl/importccl: TestImportJobEventLogging failed

ccl/importccl.TestImportJobEventLogging [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2983474&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2983474&tab=artifacts#/) on master @ [3021eec762bd1ae929becbf9694b189eb25edf86](https://github.com/cockroachdb/cockroach/commits/3021eec762bd1ae929becbf9694b189eb25edf86):


```
=== RUN   TestImportJobEventLogging
    import_stmt_test.go:7127: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestImportJobEventLogging950200406
E210514 23:26:16.380836 372664 jobs/adopt.go:260  [n1] 1  job 658520064971112449: adoption completed with error testing injected failure
    testutils.go:62: 
        	Error Trace:	testutils.go:62
        	            				soon.go:46
        	            				retry.go:197
        	            				soon.go:52
        	            				soon.go:33
        	            				testutils.go:35
        	            				import_stmt_test.go:7190
        	Error:      	Not equal: 
        	            	expected: "running"
        	            	actual  : "failed"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-running
        	            	+failed
        	Test:       	TestImportJobEventLogging
E210514 23:26:18.295108 360696 jobs/registry.go:678  [-] 2  error expiring job sessions: expire-sessions: context canceled
E210514 23:26:18.296733 360696 jobs/registry.go:683  [-] 3  failed to serve pause and cancel requests: context canceled
E210514 23:26:18.298440 361837 jobs/registry.go:678  [-] 4  error expiring job sessions: expire-sessions: context canceled
E210514 23:26:18.300097 361837 jobs/registry.go:683  [-] 5  failed to serve pause and cancel requests: context canceled
E210514 23:26:18.302730 374560 kv/kvserver/queue.go:1093  [n1,raftlog,s1,r4/1:/System{/tsd-tse}] 6  result is ambiguous (server shutdown)
E210514 23:26:18.333962 362330 jobs/registry.go:693  [-] 7  error claiming jobs: could not query jobs table: claim-jobs: context canceled
E210514 23:26:18.337622 362330 jobs/registry.go:703  [-] 8  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
    panic.go:636: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestImportJobEventLogging950200406
--- FAIL: TestImportJobEventLogging (23.31s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestImportJobEventLogging PKG=./pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
/cc @cockroachdb/bulk-io @adityamaru
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportJobEventLogging.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/importccl.TestImportJobEventLogging [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2983509&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2983509&tab=artifacts#/) on master @ [1ef6dae01af1553b8836d2626093c67cda96dbb6](https://github.com/cockroachdb/cockroach/commits/1ef6dae01af1553b8836d2626093c67cda96dbb6):


```
=== RUN   TestImportJobEventLogging
    import_stmt_test.go:7127: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestImportJobEventLogging582520553
E210514 23:54:20.328326 352657 jobs/adopt.go:260  [n1] 1  job 658525583763046401: adoption completed with error testing injected failure
    testutils.go:62: 
        	Error Trace:	testutils.go:62
        	            				soon.go:46
        	            				retry.go:197
        	            				soon.go:52
        	            				soon.go:33
        	            				testutils.go:35
        	            				import_stmt_test.go:7190
        	Error:      	Not equal: 
        	            	expected: "reverting"
        	            	actual  : "failed"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-reverting
        	            	+failed
        	Test:       	TestImportJobEventLogging
E210514 23:54:22.104050 342218 jobs/registry.go:683  [-] 2  failed to serve pause and cancel requests: could not query jobs table: cancel/pause-requested: context canceled
E210514 23:54:22.112588 342709 jobs/registry.go:678  [-] 3  error expiring job sessions: expire-sessions: context canceled
E210514 23:54:22.113839 342709 jobs/registry.go:683  [-] 4  failed to serve pause and cancel requests: context canceled
E210514 23:54:22.124223 341032 jobs/registry.go:703  [-] 5  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
    panic.go:636: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestImportJobEventLogging582520553
--- FAIL: TestImportJobEventLogging (21.51s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestImportJobEventLogging PKG=./pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
/cc @cockroachdb/bulk-io @adityamaru
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportJobEventLogging.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/importccl.TestImportJobEventLogging [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2984891&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2984891&tab=artifacts#/) on master @ [09b51b5777cad528f86372574a4f97c593e9c8bc](https://github.com/cockroachdb/cockroach/commits/09b51b5777cad528f86372574a4f97c593e9c8bc):


```
=== RUN   TestImportJobEventLogging
    import_stmt_test.go:7127: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestImportJobEventLogging410610396
E210515 05:50:55.899401 400824 jobs/adopt.go:260  [n1] 1  job 658595692330352641: adoption completed with error testing injected failure
    testutils.go:62: 
        	Error Trace:	testutils.go:62
        	            				soon.go:46
        	            				retry.go:197
        	            				soon.go:52
        	            				soon.go:33
        	            				testutils.go:35
        	            				import_stmt_test.go:7190
        	Error:      	Not equal: 
        	            	expected: "reverting"
        	            	actual  : "failed"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-reverting
        	            	+failed
        	Test:       	TestImportJobEventLogging
E210515 05:50:57.587114 389274 jobs/registry.go:693  [-] 2  error claiming jobs: could not query jobs table: claim-jobs: context canceled
E210515 05:50:57.590800 402847 kv/kvserver/consistency_queue.go:191  [n1,consistencyChecker,s1,r14/1:/Table/1{8-9}] 3  failed to send RPC: sending to all replicas failed; last error: failed to connect to n3 at 127.0.0.1:38427: stopped
E210515 05:50:57.591400 389274 jobs/registry.go:703  [-] 4  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
E210515 05:50:57.591954 402847 kv/kvserver/queue.go:1093  [n1,consistencyChecker,s1,r14/1:/Table/1{8-9}] 5  failed to send RPC: sending to all replicas failed; last error: failed to connect to n3 at 127.0.0.1:38427: stopped
E210515 05:50:57.598793 390414 jobs/registry.go:678  [-] 6  error expiring job sessions: expire-sessions: context canceled
E210515 05:50:57.600210 390414 jobs/registry.go:683  [-] 7  failed to serve pause and cancel requests: context canceled
E210515 05:50:57.621953 390935 jobs/registry.go:678  [-] 8  error expiring job sessions: expire-sessions: context canceled
E210515 05:50:57.623855 390935 jobs/registry.go:683  [-] 9  failed to serve pause and cancel requests: context canceled
    panic.go:636: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestImportJobEventLogging410610396
--- FAIL: TestImportJobEventLogging (21.80s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestImportJobEventLogging PKG=./pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
/cc @cockroachdb/bulk-io @adityamaru
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportJobEventLogging.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/importccl.TestImportJobEventLogging [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2985027&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2985027&tab=artifacts#/) on master @ [09b51b5777cad528f86372574a4f97c593e9c8bc](https://github.com/cockroachdb/cockroach/commits/09b51b5777cad528f86372574a4f97c593e9c8bc):


```
=== RUN   TestImportJobEventLogging
    import_stmt_test.go:7127: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestImportJobEventLogging727783329
E210515 08:13:28.278397 401184 jobs/adopt.go:260  [n1] 1  job 658623716416618497: adoption completed with error testing injected failure
    testutils.go:62: 
        	Error Trace:	testutils.go:62
        	            				soon.go:46
        	            				retry.go:197
        	            				soon.go:52
        	            				soon.go:33
        	            				testutils.go:35
        	            				import_stmt_test.go:7190
        	Error:      	Not equal: 
        	            	expected: "running"
        	            	actual  : "failed"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-running
        	            	+failed
        	Test:       	TestImportJobEventLogging
E210515 08:13:30.181863 390844 jobs/registry.go:678  [-] 2  error expiring job sessions: expire-sessions: context canceled
E210515 08:13:30.184512 390844 jobs/registry.go:683  [-] 3  failed to serve pause and cancel requests: context canceled
E210515 08:13:30.201033 390846 jobs/registry.go:693  [-] 4  error claiming jobs: could not query jobs table: claim-jobs: context canceled
E210515 08:13:30.203973 390846 jobs/registry.go:703  [-] 5  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
E210515 08:13:30.221515 390349 jobs/registry.go:703  [-] 6  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
E210515 08:13:30.223937 390347 jobs/registry.go:678  [-] 7  error expiring job sessions: expire-sessions: context canceled
E210515 08:13:30.225706 390347 jobs/registry.go:683  [-] 8  failed to serve pause and cancel requests: context canceled
    panic.go:636: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestImportJobEventLogging727783329
--- FAIL: TestImportJobEventLogging (23.69s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestImportJobEventLogging PKG=./pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
/cc @cockroachdb/bulk-io @adityamaru
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportJobEventLogging.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/backupccl/testutils.go
+++ b/pkg/ccl/backupccl/testutils.go
@@ -40,7 +40,8 @@ func CheckEmittedEvents(
 			t.Fatal(err)
 		}
 		foundEntry := false
-		for i, e := range entries {
+		var matchingEntryIndex int
+		for _, e := range entries {
 			if !strings.Contains(e.Message, expectedMessage) {
 				continue
 			}
@@ -56,10 +57,11 @@ func CheckEmittedEvents(
 			}
 			require.Equal(t, expectedJobType, ev.JobType)
 			require.Equal(t, jobID, ev.JobID)
-			if i >= len(expectedStatus) {
+			if matchingEntryIndex >= len(expectedStatus) {
 				return errors.New("more events fround in log than expected")
 			}
-			require.Equal(t, expectedStatus[i], ev.Status)
+			require.Equal(t, expectedStatus[matchingEntryIndex], ev.Status)
+			matchingEntryIndex++
 		}
 		if !foundEntry {
 			return errors.New("structured entry for import not found in log")

--- a/pkg/ccl/importccl/import_stmt_test.go
+++ b/pkg/ccl/importccl/import_stmt_test.go
@@ -6932,7 +6932,6 @@ func TestDetachedImport(t *testing.T) {
 
 func TestImportJobEventLogging(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	skip.WithIssue(t, 65290, "flaky test")
 	defer log.ScopeWithoutShowLogs(t).Close(t)
 
 	defer jobs.TestingSetProgressThresholds()()

